app/services/edit_defect.rb

```diff
@@ -32,6 +36,14 @@ def set_target_completion_date
 
     day, month, year = date_parts.map(&:to_i)
     date = Date.new(year, month, day)
-    defect.set_completion_date(date)
+    defect.set_target_completion_date(date)
+  end
+
+  def set_actual_completion_date
```

Comment:
I get that this is cribbed from the previous method, but it seems excessively complex, and isn't particularly readable...